### PR TITLE
Update to `cell_method` parsing

### PR DESCRIPTION
**Pull request by ukmo-ccbunney**
### Description
Iris fails to load cubes from a netCDF file if the `cell_method` attribute is malformed and throws this slightly unhelpful error:
```python
File /opt/scitools/conda/deployments/default-2024_02_20/lib/python3.11/site-packages/iris/fileformats/_nc_load_rules/helpers.py:267, in _split_cell_methods(nc_cell_methods)
    265 for ii in range(len(name_start_inds) - 1):
    266     method_indices.append((name_start_inds[ii], name_start_inds[ii + 1]))
--> 267 method_indices.append((name_start_inds[-1], len(nc_cell_methods)))
    269 # Index the string and match against each substring
    270 nc_cell_methods_matches = []

IndexError: list index out of range
```

For the netCDF files in my particular case, inspection of the stack trace reveals it was failing to correctly parse the `cell_method` attribute which was missing a space between the colon separator and time method. I.e.
```
cell_methods = "time:point"
```
rather than:
```
cell_methods = "time: point"
```

This PR makes two changes to the cell method parsing in `fileformats/_nc_load_rules/helpers.py`:
1. In `_split_cell_methods`: If the value of `cell_methods` does not match the expected `name: ` pattern, a `Warning` is issued and the function returns an empty list so cube loading can continue. This is consistent with the behaviour further down the function if the `name: value` pattern cannot be matched.

2. Updates `_CM_PARSE_NAME` and `_CM_PARSE` regexes to make the pattern matching a little more lenient by making the space after the colon separator between the `name` and `method` pair optional. The [CF Convention](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/ch07s03.html#:~:text=To%20describe%20the%20characteristic%20of%20a%20field%20that,words%20of%20the%20form%20%22%20name%3A%20method%20%22.) states that the pairs are "blank-separated list of `name: method` pairs". It _does_ show a space after the colon in the examples it gives, but the [CF Compliance checker ](https://cfchecker.ncas.ac.uk/) is happy with no space after the colon.

Unit tests have been updated to test that:
  1. Malformed `cell_method` raises a warning; e.g. name but no method, such as `cell_method='time'`
  2. Space after colon between name and method is optional; e.g. `cell_method='time:mean'`

Tested on Iris version 3.10.0.dev96.

Add any of the below labels to trigger actions on this PR:
- fixes: #6082

### CODE REVIEW

**Comment by ukmo-ccbunney** [comment]
Leaving this one as draft for the moment whilst I address some unit test failures.

**Comment by ukmo-ccbunney** [comment]
Modified slightly to address some failing unit tests.
All tests passing locally for me now.

**Comment by trexfeathers** [comment]
From @SciTools/peloton:

Related to #5165.

At the very least there needs to be an informative warning, otherwise what's the point in Iris being strict about CF.

Also, if the CF checker is OK with this, maybe we should be OK with it also? @pp-mo would love your opinion on the CF checker's place in these decisions.

**Comment by HGWright** [comment]
From @SciTools/peloton we want to accept this as CF-compliant.

**Review by ESadek-MO — CHANGES_REQUESTED** [review]
Thanks @ukmo-ccbunney, changes all look good to me! I think it's worth just adding a small whatsnew entry for this, then it'll be good to go!

**Comment by ukmo-ccbunney** [comment]
> Thanks @ukmo-ccbunney, changes all look good to me! I think it's worth just adding a small whatsnew entry for this, then it'll be good to go!

Done.

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -52,6 +52,9 @@ This document explains the changes made to Iris for this release
 #. `@rcomer`_ enabled partial collapse of multi-dimensional string coordinates,
    fixing :issue:`3653`. (:pull:`5955`)
 
+#. `@ukmo-ccbunney`_ improved error handling for malformed `cell_method`
+   attribute. Also made cell_method string parsing more lenient w.r.t.
+   whitespace. (:pull:`6181`)
```
Comment (ESadek-MO) [inline docs/src/whatsnew/latest.rst:57]:
```suggestion
   whitespace. (:pull:`6083`)
```

**Review by ESadek-MO — CHANGES_REQUESTED** [review]
Wrong PR number!

File: docs/src/whatsnew/latest.rst (hunk 1 above)
Comment (ukmo-ccbunney) [reply docs/src/whatsnew/latest.rst]: Whoops! Good spot 🙄

**Review by ESadek-MO — APPROVED** [review]

**Pull request merged**